portal/patient/fwk/libs/verysimple/DB/DataDriver/MySQLi.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/** @package verysimple::DB::DataDriver */
+require_once ("IDataDriver.php");
+require_once ("verysimple/DB/ISqlFunction.php");
+require_once ("verysimple/DB/DatabaseException.php");
+require_once ("verysimple/DB/DatabaseConfig.php");
+
+/**
+ * An implementation of IDataDriver that communicates with
+ * a MySQL server.
+ * This is one of the native drivers
+ * supported by Phreeze
+ *
+ * @package verysimple::DB::DataDriver
+ * @author VerySimple Inc. <noreply@verysimple.com>
+ * @copyright 1997-2010 VerySimple Inc.
+ * @license http://www.gnu.org/licenses/lgpl.html LGPL
+ * @version 1.0
+ */
+class DataDriverMySQLi implements IDataDriver {
+	/** @var characters that will be escaped */
+	static $BAD_CHARS = array (
+			"\\",
+			"\0",
+			"\n",
+			"\r",
+			"\x1a",
+			"'",
+			'"' 
+	);
+	
+	/** @var characters that will be used to replace bad chars */
+	static $GOOD_CHARS = array (
+			"\\\\",
+			"\\0",
+			"\\n",
+			"\\r",
+			"\Z",
+			"\'",
+			'\"' 
+	);
+	
+	/**
+	 * @inheritdocs
+	 */
+	function GetServerType() {
+		return "MySQLi";
+	}
+	function Ping($connection) {
+		return mysqli_ping ( $connection );
+	}
+	
+	/**
+	 * @inheritdocs
+	 */
+	function Open($connectionstring, $database, $username, $password, $charset = '', $bootstrap = '') {
+		if (! function_exists ( "mysqli_connect" ))
+			throw new DatabaseException ( 'mysqli extension is not enabled on this server.', DatabaseException::$CONNECTION_ERROR );
+			
+			// if the port is provided in the connection string then strip it out and provide it as a separate param
+		$hostAndPort = explode ( ":", $connectionstring );
+		$host = $hostAndPort [0];
+		$port = count ( $hostAndPort ) > 1 ? $hostAndPort [1] : null;
+		
+		$connection = mysqli_connect ( $host, $username, $password, $database, $port );
+		
+		if (mysqli_connect_errno ()) {
+			throw new DatabaseException ( "Error connecting to database: " . mysqli_connect_error (), DatabaseException::$CONNECTION_ERROR );
+		}
+		
+		if ($charset) {
+			mysqli_set_charset ( $connection, $charset );
```

Comment:
TODO:
I need to look at this function to ensure it's doing SET NAMES UTF8
(github search not working, so will do better searching at home)